desc/objectives/objective_funs.py

```diff
@@ -130,10 +158,44 @@ def build(self, use_jit=None, verbose=1):
             self._scalar = False
 
         self._set_derivatives()
+        sub_obj_jac_chunk_sizes = [obj._jac_chunk_size for obj in self.objectives]
+        warnif(
+            np.any(sub_obj_jac_chunk_sizes) and self._deriv_mode != "blocked",
+            UserWarning,
+            "'jac_chunk_size' was passed into one or more sub-objectives, but the"
+            "ObjectiveFunction is  using 'batched' deriv_mode, so sub-objective "
+            "'jac_chunk_size' will be ignored in favor of the ObjectiveFunction's "
+            f"'jac_chunk_size' of {self._jac_chunk_size}."
+            " Specify 'blocked' deriv_mode if each sub-objective is desired to have a "
+            "different 'jac_chunk_size' for its Jacobian computation.",
+        )
+        warnif(
+            self._jac_chunk_size is not None and self._deriv_mode == "blocked",
+            UserWarning,
+            "'jac_chunk_size' was passed into ObjectiveFunction, but the"
+            "ObjectiveFunction is using 'blocked' deriv_mode, so sub-objective "
+            "'jac_chunk_size' are used to compute each sub-objective's Jacobian, "
+            "`ignoring the ObjectiveFunction's 'jac_chunk_size'.",
+        )
+
         if not self.use_jit:
             self._unjit()
 
         self._set_things()
+        if self._jac_chunk_size is None and self._deriv_mode == "batched":
+            # set jac_chunk_size to 1/4 of number columns of Jacobian
+            # as the default for batched deriv_mode
+            self._jac_chunk_size = int(np.ceil(self.dim_x / 4))
```

Comment:
To make sure I understand: The existing master code when using `deriv_mode="batched"` is equivalent to `jac_chunk_size=dim.x`, correct? So why change the default to be one-fourth of that? Did you find that size is generally the best trade-off between speed and memory? Just want to check since I think this will change the behavior of old code.

Reply:
yea, Rory/Yigit wanted something auto to be better. I should check dim_x and current avail. memory and do it from that, but have not done a study to get a heurstic memory use vs chunk_size and dim_x yet

Reply:
I'd prefer to keep `chunk_size=None` be no chunking (`chunk_size=dim_x`). We could add a `chunk_size="auto"` option that for now uses dim_x/4 until we come up with a better heuristic (or maybe "auto" means chunk_size=1000 or something which is probably a safer bet than dim_x/4)